kpack-image-builder/controllers/imageprocessfetcher/image_process_fetcher.go

```diff
@@ -66,8 +65,11 @@ func (f *ImageProcessFetcher) Fetch(imageRef string, credsOption remote.Option,
 
 // Reconstruct command with arguments into a single command string
 func extractFullCommand(process launch.Process) string {
-	commandWithArgs := append([]string{process.Command}, process.Args...)
-	return strings.Join(commandWithArgs, " ")
+	cmdString := process.Command
+	for _, a := range process.Args {
+		cmdString = fmt.Sprintf(`%s %q`, cmdString, a)
```

Comment:
I wonder if this is resilient to all arg content. Is it possible that a valid arg could contain quoting that would become invalid when wrapped in further quotes?

Reply:
We should expect any valid args would be properly quoted and escaped. It is possible that manifest commands could come in without escaping, but there wouldn't be any args, so the quoting shouldn't come into play.

Regardless, this is more of a band-aid fix to enable buildpacks that provide a `bash -c` style cmd and args. We should avoid string processing of this sort whenever possible. We should have discussion around how to move forward.

Reply:
I think Kieron's concern is that, by wrapping a string with quotes without escaping the string being wrapped, something like `"properly \"escaped\" string"` becomes `""properly \"escaped\" string""`, which is not the same as `"\"properly \\\"escaped\\\" string\""`.

Reply:
Looking at [fmt](https://pkg.go.dev/fmt) docs, `%q	a single-quoted character literal safely escaped with Go syntax.` I think we should be safe.

Reply:
I just tried this out in the Go playground and it looks to be doing what we want it to: https://go.dev/play/p/UZnII9jxCZe